js/main.js

```diff
@@ -137,14 +137,52 @@ function handleMailValidation(input, msg) {
   const inputVal = input.value.trim().toLowerCase();
   if (inputVal !== input.value.trim()) {
     displayErr(input, msg);
-  } else {
-    displayErr(input, '');
-    form.submit();
+    return false;
   }
+  displayErr(input, '');
+  return true;
+}
+
+// local Storage
+
+function createUserData(formElement) {
+  return {
+    username: formElement.elements.username.value.trim(),
+    email: formElement.elements.email.value.trim(),
+    message: formElement.elements.message.value.trim(),
+  };
 }
 
+function storeInfo(formElement) {
+  localStorage.setItem('userInfo', JSON.stringify(createUserData(formElement)));
+}
+
+// submit form
 form.addEventListener('submit', (e) => {
   e.preventDefault();
 
-  handleMailValidation(form.elements.email, MAIL_ERROR);
+  const submit = handleMailValidation(form.elements.email, MAIL_ERROR);
+
+  if (submit) {
+    storeInfo(form);
+    form.submit();
+  }
```

Comment:
- [ ] Following the project requirement, after filling the form, and not submitting it, the entered values should be saved in the browser (localStorage). If the user reloads the page or closes the browser and opens it again, the form inputs should be auto-populated with the values entered in the previous visit to the page.

Reply:
Hello amine, I really appreciate your feedback.
Could I ask you when would the input values get stored in the local storage?
should we store the information while the user is typing or should we store it when the user has finished typing.

Reply:
Hi @NabilHY,
It would be better to store the information while the user is typing. You can use the eventListener  `keydown` to implement that.